torch/nested/__init__.py

```diff
@@ -302,3 +304,87 @@ def narrow(tensor: Tensor, dim: int, start: Union[int, Tensor], length: Union[in
         raise RuntimeError(f"Specified layout is unsupported for nested narrow: {layout}")
 
     return nt
+
+
+def nested_tensor_from_jagged(
+    values: Tensor,
+    offsets: Optional[Tensor] = None,
+    lengths: Optional[Tensor] = None,
+    jagged_dim: Optional[int] = None,
+) -> Tensor:
+    r"""
+Constructs a jagged layout nested tensor from the given jagged components. The jagged layout
+consists of a required values buffer with the jagged dimension collapsed into a single dimension.
```

Comment:
nit: collapsed -> packed? (here and other places) collapsed feels like reduced... but maybe that is just me

Reply:
yeah packed is better! will change